functions/signing/signing.py

```diff
@@ -0,0 +1,244 @@
+
+        # use openssl to sign the content
+        sha384_file = file_to_sign + '.sha384'
+        sign_com = 'openssl dgst -sha384 -sign {0} -out {1} {2}'.format(pem_file, sha384_file, file_to_sign)
+        command = shlex.split(sign_com)
+        com = Popen(command, shell=False, stdin=PIPE, stdout=PIPE, stderr=PIPE)
+        out, err = com.communicate()
+
+        if err:
+            raise Exception(err)
+
+        # base64 encode the signature
+        file_name_without_extension = os.path.splitext(file_to_sign)[0]
+        sig_file_name = '{}.sig'.format(file_name_without_extension)
+        sign_com = 'openssl base64 -in {0} -out {1}'.format(sha384_file, sig_file_name)
+        command = shlex.split(sign_com)
+        com = Popen(command, shell=False, stdin=PIPE, stdout=PIPE, stderr=PIPE)
+        out, err = com.communicate()
+
+        if err:
+            raise Exception(err)
+
+        # get the base64 encoded signature
+        with codecs.open(sig_file_name, 'r', encoding='utf-8') as in_file:
+            signed_content = in_file.read()
+
+        # save the signed content
+        file_content = []
+        signature = {'si': 'uW', 'sig': signed_content}
+        file_content.append(signature)
+        write_file(sig_file_name, file_content)
+
+        return sig_file_name
+
+    @staticmethod
+    def verify_signature(content_file, sig_file, pem_file=None):
+        """
+        Verify that the file content has not changed since it was signed
+        :param str|unicode content_file:
+        :param str|unicode sig_file:
+        :param str|unicode|None pem_file:
+        :return:
+        """
+
+        temp_dir = tempfile.mkdtemp(prefix='tempVerify_')
+
+        try:
+
+            # if pem file was not passed, use the default one
+            if not pem_file:
+                pem_file = os.path.join(temp_dir, 'uW-vk.pem')
+                download_file('https://pki.unfoldingword.org/uW-vk.pem', pem_file)
+
+            # get the uW signature from the sig file
+            with codecs.open(sig_file, 'r', 'utf-8-sig') as in_file:
+                sig_file_content = json.loads(in_file.read())
+
+            signature = [x['sig'] for x in sig_file_content if x['si'] == 'uW'][0]
+            signature_path = os.path.join(temp_dir, 'signature.sig')
+
+            # save the signature to a temp file
+            with open(signature_path, str('w')) as out_file:
+                out_file.write(b64decode(signature))
+
+            # Use openssl to verify signature
+            command_str = 'openssl dgst -sha384 -verify {0} -signature {1} {2}'.format(pem_file, signature_path,
+                                                                                       content_file)
+            command = shlex.split(command_str)
+            com = Popen(command, shell=False, stdin=PIPE, stdout=PIPE, stderr=PIPE)
+            out, err = com.communicate()
+
+            return out.strip() == 'Verified OK'
```

Comment:
Is possible to determine success based on the return code, instead of parsing stdout? I'm worried that if `openssl` ever changes its dialog, this check won't work anymore.